File: Algorithms/Implementation/Easy/Modified Kaprekar Numbers/cpp/kaprekar1/kaprekar.cpp
#include <cmath>
#include <iostream>

void kaprekarNumbers(int p, int q) {
	bool none = true;

	for (int i = p; i <= q; i++) {
		int d = std::to_string(i).length();
		std::string square = std::to_string(static_cast<long>(pow(i, 2)));
		int l = square.length() > d ? std::atoi(square.substr(0, square.length() - d).c_str()) : 0;
		int r = std::atoi(square.substr(square.length() - d).c_str());

		if (l + r == i) {
			none = false;
			std::cout << i << (i == q ? "" : " ");
		}
	}

	if (none) {
		std::cout << "INVALID RANGE";
	}
}

int main()
{
	int p = 0, q = 0;
	std::cin >> p >> q;
	kaprekarNumbers(p, q);
}

File: Algorithms/Implementation/Easy/Modified Kaprekar Numbers/java/kaprekar1/Kaprekar.java
import java.util.Arrays;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;


class Kaprekar {
	private static void kaprekarNumbers(int p, int q) {
		Integer[] kaprekar_numbers = new Integer[0];

		for (int i = p; i <= q; i++) {
			int d = Integer.toString(i).length();
			String square = String.valueOf((long)Math.pow(i, 2));
			int l = square.length() > d ?
					Integer.parseInt(square.substring(0, square.length() - d)) : 0;
			int r = Integer.parseInt(square.substring(square.length() - d));
			
			if (l + r == i) {
				List<Integer> list = new ArrayList<>(Arrays.asList(kaprekar_numbers));
				list.add(i);
				kaprekar_numbers = list.toArray(kaprekar_numbers);
			}
		}

		if (kaprekar_numbers.length > 0) {
			for (int i = 0; i < kaprekar_numbers.length; i++) {
				System.out.print(i == kaprekar_numbers.length - 1 ?
						kaprekar_numbers[i] : kaprekar_numbers[i] + " ");
			}
		} else {
			System.out.println("INVALID RANGE");
		}
	}

	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		int p = input.nextInt();
		int q = input.nextInt();
		kaprekarNumbers(p, q);
	}
}